consensus/parlia/parlia.go

```diff
@@ -1223,6 +1223,15 @@ func (p *Parlia) FinalizeAndAssemble(chain consensus.ChainHeaderReader, header *
 		}
 	}
 
+	// consider withdrawals
+	shanghai := chain.Config().IsShanghai(header.Number, header.Time)
+	if shanghai {
+		// All blocks after Shanghai must include a withdrawals root.
+		if header.WithdrawalsHash == nil {
+			header.WithdrawalsHash = new(common.Hash)
```

Comment:
The `WithdrawalsHash` decode issue will be fixed by another PR. And you cannot change Shanghai's header field value, because it has been enabled on mainnet.